components/AddExercises/ExerciseList.jsx

```diff
@@ -36,92 +39,111 @@ const Buttontext = styled.Text`
-  }, {});
-  return bucketData;
-};
+const SortByButton = styled(SortByPopup)`
+  position: absolute;
+  right: 15px;
+  top: 22px;
+  height: 35px;
+  width: 35px;
+  margin: 0px 15px 0px 0px;
+`;
 
 const renderHeader = ({ section }) => (
   <SectionHeader>{section.title}</SectionHeader>
 );
 
-const ExerciseList = (props) => {
-  const { items, onExercisesAdd } = props;
-
-  const parsedItems = parseItems(items);
-  const [search, setSearch] = React.useState('');
-  const [filteredDataSource, setFilteredDataSource] = React.useState(parsedItems);
-  const [masterDataSource, setMasterDataSource] = React.useState(parsedItems);
-  const [exerciseCount, setExerciseCount] = React.useState(0);
-  const [addedExercises] = React.useState([]);
+const ExerciseList = ({ onExercisesAdd, parsedItemsName, parsedItemsMuscleGroups }) => {
+  const [dataState, setDataState] = useState({ isSortByMuscleGroup: false, filteredDataSource: [], masterDataSource: parsedItemsName });
   const navigation = useNavigation();
+  const [search, setSearch] = useState('');
+  const [exerciseCount, setExerciseCount] = useState(0);
+  const [addedExercises] = useState([]);
+
+  const handleSortByNamePress = () => {
+    setDataState({ isSortByMuscleGroup: false, filteredDataSource: parsedItemsName, masterDataSource: parsedItemsName });
+  };
+
+  const handleSortByMuscleGroupPress = () => {
+    setDataState({ isSortByMuscleGroup: true, filteredDataSource: parsedItemsMuscleGroups, masterDataSource: parsedItemsMuscleGroups });
+  };
 
   const searchFilterFunction = (text) => {
-    // Check if searched text is not blank
     if (text) {
-      // Inserted text is not blank
-      // Filter the masterDataSource
-      // Update FilteredDataSource
-      const firstLetter = text[0].toUpperCase();
-      if (masterDataSource[firstLetter]) {
-        const newData = masterDataSource[firstLetter].filter((item) => {
+      const filteredItems = [];
+      const textData = text.toUpperCase();
+      Object.keys(dataState.masterDataSource).forEach((key) => {
+        const newData = dataState.masterDataSource[key].filter((item) => {
           const itemData = item.name
             ? item.name.toUpperCase()
             : '';
-          const textData = text.toUpperCase();
           return itemData.indexOf(textData) > -1;
         });
-        const filteredData = {};
-        filteredData[firstLetter] = newData;
-        setFilteredDataSource(filteredData);
-        setSearch(text);
+        if (newData) {
+          Object.keys(newData).forEach((each) => {
+            if (newData[each]) {
+              filteredItems.push(newData[each]);
+            }
+          });
+        }
+        filteredItems.sort((a, b) => a.name.localeCompare(b.name));
+      });
+      setDataState({ ...dataState, filteredDataSource: filteredItems });
+      setSearch(text);
+    } else {
+      setDataState({ ...dataState, filteredDataSource: dataState.masterDataSource });
+      setSearch(text);
```

Comment:
Consider passing in the original exerciseList and then just filtering that list so you don't have to deal with this as an object